### PR TITLE
Add GRS attributes to pi volume group data sources and related

### PR DESCRIPTION
**Pull request by Alexander-Kita**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

pi_volume_group (d):

```
=== RUN   TestAccIBMPIVolumeGroupDataSource_basic
--- PASS: TestAccIBMPIVolumeGroupDataSource_basic (10.44s)
PASS
```
pi_volume_groups (d):

```
=== RUN   TestAccIBMPIVolumeGroupsDataSource_basic
--- PASS: TestAccIBMPIVolumeGroupsDataSource_basic (13.01s)
PASS
```

pi_volume_group_details (d):

```
=== RUN   TestAccIBMPIVolumeGroupDetailsDataSource_basic
--- PASS: TestAccIBMPIVolumeGroupDetailsDataSource_basic (8.56s)
PASS
```

pi_volume_groups_details (d):

```
=== RUN   TestAccIBMPIVolumeGroupsDetailsDataSource_basic
--- PASS: TestAccIBMPIVolumeGroupsDetailsDataSource_basic (15.04s)
PASS
```

pi_volume_group (r):

```
=== RUN   TestAccIBMPIVolumeGroupUpdate
--- PASS: TestAccIBMPIVolumeGroupUpdate (355.05s)
PASS
```

Description: 
- Adds attributes storage_pool, replication_sites, and auxiliary as attributes to pi_volume_group, pi_volume_groups, pi_volume_group_details, and pi_volume_groups_details data sources.

- Adds replication_sites to pi_volume_group resource

- Adds status_description_errors attribute into schema (missing previously)

### CODE REVIEW

File: ibm/service/power/data_source_ibm_pi_volume_group.go [hunk 1]
```diff
@@ -33,11 +33,22 @@ func DataSourceIBMPIVolumeGroup() *schema.Resource {
...
 			Attr_ConsistencyGroupName: {
 				Computed:    true,
 				Description: "The name of consistency group at storage controller level.",
 				Type:        schema.TypeString,
 			},
+			Attr_ReplicationSites: {
+				Computed:    true,
+				Description: "List of Replication Sites.",
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
```
Comment (michaelkad) [inline ibm/service/power/data_source_ibm_pi_volume_group.go:50]:
```suggestion
			Attr_ReplicationSites: {
				Computed:    true,
				Description: "List of Replication Sites.",
				Elem:        &schema.Schema{Type: schema.TypeString},
				Type:        schema.TypeList,
```

File: ibm/service/power/data_source_ibm_pi_volume_group_details.go [hunk 2]
```diff
@@ -42,6 +47,12 @@ func DataSourceIBMPIVolumeGroupDetails() *schema.Resource {
 				Description: "The replication status of volume group.",
 				Type:        schema.TypeString,
 			},
+			Attr_ReplicationSites: {
+				Computed:    true,
+				Description: "List of Replication Sites.",
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
```
Comment (michaelkad) [inline ibm/service/power/data_source_ibm_pi_volume_group_details.go:55]: Could you sort them in an alphabetical order?

File: ibm/service/power/data_source_ibm_pi_volume_groups.go [hunk 3]
```diff
@@ -50,6 +55,12 @@ func DataSourceIBMPIVolumeGroups() *schema.Resource {
 							Description: "The replication status of volume group.",
 							Type:        schema.TypeString,
 						},
+						Attr_ReplicationSites: {
+							Computed:    true,
+							Description: "List of Replication Sites.",
+							Type:        schema.TypeList,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
```
Comment (michaelkad) [inline ibm/service/power/data_source_ibm_pi_volume_groups.go:63]: same as above

File: ibm/service/power/data_source_ibm_pi_volume_groups_details.go [hunk 4]
```diff
@@ -50,11 +55,22 @@ func DataSourceIBMPIVolumeGroupsDetails() *schema.Resource {
 							Description: "The replication status of volume group.",
 							Type:        schema.TypeString,
 						},
+						Attr_ReplicationSites: {
+							Computed:    true,
+							Description: "List of Replication Sites.",
+							Type:        schema.TypeList,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
```
Comment (michaelkad) [inline ibm/service/power/data_source_ibm_pi_volume_groups_details.go:63]: same as above.

File: website/docs/d/pi_volume_group_details.html.markdown [hunk 5]
```diff
@@ -42,15 +42,18 @@ Review the argument references that you can specify for your data source.
 ## Attribute reference
 In addition to all argument reference list, you can access the following attribute references after your data source is created. 
 
+- `auxiliary` - (Boolean) Indicates if the volume group is auxiliary.
 - `consistency_group_name` - (String) The name of consistency group at storage controller level.
 - `id` - (String) The unique identifier of the volume group.
 - `replication_status` - (String) The replication status of volume group.
+- `replication_sites` - ([]String) List of replication sites for the volume group.
```
Comment (michaelkad) [inline website/docs/d/pi_volume_group_details.html.markdown:49]:
```suggestion
- `replication_sites` - (List) List of replication sites for the volume group.
```

File: website/docs/d/pi_volume_groups.html.markdown [hunk 6]
```diff
@@ -43,14 +43,17 @@ In addition to all argument reference list, you can access the following attribu
 - `volume_groups`: List of all volume groups.
   
   Nested scheme for `volume_groups`:
+  - `auxiliary` - (Boolean) Indicates if the volume group is auxiliary.
   - `consistency_group_name` - (String) The name of consistency group at storage controller level.
   - `id` - (String) The unique identifier of the volume group.
   - `replication_status` - (String) The replication status of volume group.
+  - `replication_sites` - ([]String) List of replication sites for the volume group.
```
Comment (michaelkad) [inline website/docs/d/pi_volume_groups.html.markdown:50]:
```suggestion
  - `replication_sites` - (List) List of replication sites for the volume group.
```

File: website/docs/d/pi_volume_groups_details.html.markdown [hunk 7]
```diff
@@ -43,15 +43,18 @@ In addition to all argument reference list, you can access the following attribu
 - `volume_groups` - (List) List of all volume group.
   
   Nested scheme for `volume_groups`:
+  - `auxiliary` - (Boolean) Indicates if the volume group is auxiliary.
   - `consistency_group_name` - (String) The name of consistency group at storage controller level.
   - `id` - (String) The unique identifier of the volume group.
   - `replication_status` - (String) The replication status of volume group.
+  - `replication_sites` - ([]String) List of replication sites for the volume group.
```
Comment (michaelkad) [inline website/docs/d/pi_volume_groups_details.html.markdown:50]:
```suggestion
  - `replication_sites` - (List) List of replication sites for the volume group.
```

File: website/docs/r/pi_volume_group.html.markdown [hunk 8]
```diff
@@ -55,10 +55,21 @@ Review the argument references that you can specify for your resource.
 In addition to all argument reference list, you can access the following attribute reference after your resource is created.
 
 - `id` - (String) The unique identifier of the volume group. The ID is composed of `<pi_cloud_instance_id>/<volume_group_id>`.
+- `auxiliary` - (Boolean) Indicates if the volume group is auxiliary.
 - `consistency_group_name` - (String) The consistency Group Name if volume is a part of volume group.
+- `replication_sites` - ([]String) List of replication sites for the volume group.
```
Comment (michaelkad) [inline website/docs/r/pi_volume_group.html.markdown:60]:
```suggestion
- `replication_sites` - (List) List of replication sites for the volume group.
```

**Review by michaelkad — CHANGES_REQUESTED** [review]
Few comments

File: website/docs/d/pi_volume_group_details.html.markdown [hunk 9]
```diff
@@ -42,15 +42,18 @@ Review the argument references that you can specify for your data source.
...
 - `replication_status` - (String) The replication status of volume group.
+- `replication_sites` - (List) List of replication sites for the volume group.
 - `status` - (String) The status of the volume group.
 - `status_description_errors` - (List) The status details of the volume group.
 
   Nested scheme for `status_description_errors`:
   - `key` - (String) The volume group error key.
   - `message` - (String) The failure message providing more details about the error key.
   - `volume_ids` - (List) List of volume IDs, which failed to be added/removed to/from the volume group, with the given error.
+- `storage_pool` - (String) Storage pool the volume group.
```
Comment (michaelkad) [inline website/docs/d/pi_volume_group_details.html.markdown:57]:
```suggestion
- `storage_pool` - (String) Storage pool of the volume group.
```

File: website/docs/d/pi_volume_groups.html.markdown [hunk 10]
```diff
@@ -43,14 +43,17 @@ In addition to all argument reference list, you can access the following attribu
...
   - `replication_status` - (String) The replication status of volume group.
+  - `replication_sites` - (List) List of replication sites for the volume group.
   - `status` - (String) The status of the volume group.
   - `status_description_errors` - (List) The status details of the volume group.
 
       Nested scheme for `status_description_errors`:
       - `key` - (String) The volume group error key.
       - `message` - (String) The failure message providing more details about the error key.
       - `volume_ids` - (List) List of volume IDs, which failed to be added/removed to/from the volume group, with the given error.
+  - `storage_pool` - (String) Storage pool the volume group.
```
Comment (michaelkad) [inline website/docs/d/pi_volume_groups.html.markdown:58]:
```suggestion
  - `storage_pool` - (String) Storage pool of the volume group.
```

File: website/docs/d/pi_volume_groups_details.html.markdown [hunk 11]
```diff
@@ -43,15 +43,18 @@ In addition to all argument reference list, you can access the following attribu
...
   - `replication_status` - (String) The replication status of volume group.
+  - `replication_sites` - (List) List of replication sites for the volume group.
   - `status` - (String) The status of the volume group.
   - `status_description_errors` - (List) The status details of the volume group.
 
       Nested scheme for `status_description_errors`:
       - `key` - (String) The volume group error key.
       - `message` - (String) The failure message providing more details about the error key.
       - `volume_ids` - (List) List of volume IDs, which failed to be added/removed to/from the volume group, with the given error.
+  - `storage_pool` - (String) Storage pool the volume group.
```
Comment (michaelkad) [inline website/docs/d/pi_volume_groups_details.html.markdown:58]:
```suggestion
  - `storage_pool` - (String) Storage pool of the volume group.
```

File: website/docs/r/pi_volume_group.html.markdown [hunk 12]
```diff
@@ -55,10 +55,21 @@ Review the argument references that you can specify for your resource.
...
+- `replication_sites` - (List) List of replication sites for the volume group.
 - `replication_status` - (String) The replication status of volume group.
+- `status` - (String) The status of the volume group.
+- `status_description_errors` - (List) The status details of the volume group.
+
+      Nested scheme for `status_description_errors`:
+      - `key` - (String) The volume group error key.
+      - `message` - (String) The failure message providing more details about the error key.
+      - `volume_ids` - (List) List of volume IDs, which failed to be added/removed to/from the volume group, with the given error.
+- `storage_pool` - (String) Storage pool the volume group.
```
Comment (michaelkad) [inline website/docs/r/pi_volume_group.html.markdown:69]:
```suggestion
- `storage_pool` - (String) Storage pool of the volume group.
```

**Review by michaelkad — COMMENTED** [review]
Few more comments.

**Review by michaelkad — CHANGES_REQUESTED** [review]
Update  for `resource_ibm_pi_volume_group` are missing

**Review by michaelkad — APPROVED** [review]

**Review by ismirlia — APPROVED** [review]

**Comment by Alexander-Kita** [comment]
@michaelkad Re-requesting review for replication sites attribute checking

File: website/docs/r/pi_volume_group.html.markdown [hunk 13]
```diff
@@ -55,10 +55,21 @@ Review the argument references that you can specify for your resource.
...
+- `status_description_errors` - (List) The status details of the volume group.
+
+      Nested scheme for `status_description_errors`:
+      - `key` - (String) The volume group error key.
+      - `message` - (String) The failure message providing more details about the error key.
+      - `volume_ids` - (List) List of volume IDs, which failed to be added/removed to/from the volume group, with the given error.
+- `storage_pool` - (String) Storage pool of the volume group.
 - `volume_group_id` - (String) The unique identifier of the volume group.
 - `volume_group_status` - (String) The status of the volume group.
+- `volume_group_name` - (String) The name of the volume group.
```
Comment (michaelkad) [inline website/docs/r/pi_volume_group.html.markdown:72]: I don't think this is needed or set in the code.

**Review by michaelkad — APPROVED** [review]
lgtm

**Review by michaelkad — APPROVED** [review]

**Comment by Alexander-Kita** [comment]
Opened in main repo.

**Pull request closed without merging**